src/bin/pg_dump/dump_babel_utils.c

```diff
@@ -190,6 +190,57 @@ fixTsqlTableTypeDependency(Archive *fout, DumpableObject *dobj, DumpableObject *
 		addObjectDependency(dobj, tytable->dobj.dumpId);
 }
 
+/*
+ * In Babelfish v1.2, we redefined some operators to fix issues.
+ * However, we cannot replace those operators using upgrade scripts
+ * because customer-defined objects can depend on the operators.
+ * This function will do in-place substitutions as a part of pg_dump.
+ */
+void
+fixOprRegProc(Archive *fout,
+			  const OprInfo *oprinfo,
+			  const char *oprleft,
+			  const char *oprright,
+			  char **oprregproc)
+{
+	const char *nsname;
+	const char *oprname;
+
+	if (!isBabelfishDatabase(fout) || fout->remoteVersion >= 140000)
+		return;
+
+	nsname = oprinfo->dobj.namespace->dobj.name;
+	if (strcmp(nsname, "sys") != 0)
+		return;
+
+	oprname = oprinfo->dobj.name;
+	if (strcmp(oprname, "+") == 0 &&
+		strcmp(oprleft, "\"text\"") == 0 &&
```

Comment:
are we sure that these are the only possible operators we can have? just wondering if we can have text -> sys.text -> pg_catalog.text? Basically how do we make sure that we have an operator with sys or without sys or both? If it is completely dependent on definition in sql script then we are good.

Reply:
Here, `"text"` is derived from `PQgetvalue()` and the same as `pg_catalog.text`. If we have `sys.text` in the future, it will return `"sys"."text"`.

Reply:
Here, `"text"` is derived from `PQgetvalue()` and the same as `pg_catalog.text`. If we have `sys.text` in the future, it will return `"sys"."text"`.